clients/src/test/java/org/apache/kafka/clients/consumer/internals/MembershipManagerImplTest.java

```diff
@@ -220,6 +233,7 @@ public void testTransitionToReconcilingIfEmptyAssignmentReceived() {
 
     @Test
     public void testMemberIdAndEpochResetOnFencedMembers() {
+        createCommitRequestManager(false);
```

Comment:
we shouldn't need an actual `commitRequestManager` here, where we don't need anything specific related to commits and a mock should do. I guess you could remove it and just return a completed future for the autoCommit before revocation :

`when(commitRequestManager.maybeAutoCommitSyncBeforeRevocation(anyLong())).thenReturn(CompletableFuture.completedFuture(null));`

Probably doing it once inside the `createMemberInStableState` will allow you to remove most of the createCommitRequestManager calls